sdk/src/transaction_context.rs

```diff
@@ -408,22 +419,17 @@ pub struct InstructionContext {
 }
 
 impl InstructionContext {
-    /// New
+    /// Used together with TransactionContext::get_next_instruction_context()
     #[cfg(not(target_os = "solana"))]
-    fn new(
-        nesting_level: usize,
-        instruction_accounts_lamport_sum: u128,
-        program_accounts: Vec<usize>,
-        instruction_accounts: Vec<InstructionAccount>,
-        instruction_data: Vec<u8>,
-    ) -> Self {
-        InstructionContext {
-            nesting_level,
-            instruction_accounts_lamport_sum,
-            program_accounts,
-            instruction_accounts,
-            instruction_data,
-        }
+    pub fn configure(
```

Comment:
First time I'm looking at this corner of the codebase so forgive the kinda long winded way to get to my point 😅

So the intention of this change is that during CPI, programs won't have to create an instruction context in vm memory and then CPI will have to copy it to host memory. Programs will write directly to the "next" instruction context, passed in as a writable region, before doing CPI. That part isn't done yet, but it will not use `InstructionContext::configure` right?

If the assumption above is correct, then I think that the arguments to `configure` here should be owned values, not references. In general when something requires ownership, it should communicate so at the highest possible layer, so callers can (if they have the option), skip a copy by giving up their own values. For example it looks like `MessageProcessor::process_message` can move `instruction_accounts` without copying.

Reply:
I probably should have mentioned it in this PR description as well: There will be another followup, which I am currently working on, that "inlines" all of the `instruction_trace` allocations and its nested allocations in `InstructionContext` into one consecutive allocation. That won't have `Vec`s for the fields anymore so `InstructionContext::configure()` will have to copy into them anyway.

And you are right, the top level instructions could go without a copy of the `InstructionContext` parameters if the transaction loading code would be refactored as well. But, I don't worry about that for now. Saving allocations is necessary for ABIv2 and will already help performance in itself.

Reply:
I see! Alright this looks good then